### PR TITLE
feat(prospects): add event rule for prospects

### PR DESCRIPTION
**Pull request by jpetto**
## Goal

create event bridge rule for prospects coming in from ML processes.

this is PHASE 1 of this change. deploying this to production will not have any impact until we update ML systems to send prospects to event bridge instead of SQS - but this can (and should!) still go out to prod!

PHASE 2 will satisfy the `TODO`s left in this PR - the effect of which will be any prospects sent to the prod event bridge will be sent to the dev event bridge (which will in turn send them to the dev sqs queue and the dev curation admin tools prospect db).

this has successfully been deployed to dev and tested there. it successfully sends events down to the dev sqs queue and i'm seeing the logs in the dev lambda!

## References

JIRA ticket:
* https://getpocket.atlassian.net/browse/BACK-1412

### CODE REVIEW

File: .aws/src/events-schema/prospectEvents.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// TODO: do we need this file? ./userEvents.ts is imported in ../main.ts, but is never
```
Comment (sri-kirsh) [inline .aws/src/events-schema/prospectEvents.ts:1]: this folder will have the schema bindings generated by the schema-registry console..the dev has to manually generate it and copy-paste it here 

the consuming application can copy this code into their repository, e.g https://github.com/Pocket/account-data-deleter/blob/main/lambda/events/schemas/accountDeleteEvent.ts

File: .aws/src/event-rules/user-api-events/userApiEventRules.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ export class UserApiEvents extends Resource {
         name: `${config.prefix}-UserEvents-Rule`,
         pattern: {
           source: [eventConfig.source],
-          'detail-type': [...eventConfig.detailType],
+          'detail-type': eventConfig.detailType,
```
Comment (jpetto) [inline .aws/src/event-rules/user-api-events/userApiEventRules.ts:46]: this is already an array in the config

File: .aws/src/event-rules/user-api-events/userApiEventRules.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ export class UserApiEvents extends Resource {
 
     return new PocketEventBridgeRuleWithMultipleTargets(
       this,
-      `${config.prefix}-EventBridge-Rule`,
+      `${config.prefix}-User-Api-EventBridge-Rule`,
```
Comment (jpetto) [inline .aws/src/event-rules/user-api-events/userApiEventRules.ts:62]: these rules should have unique names

**Review by kschelonka — APPROVED** [review]
This seems fine as written for prod, but I'm concerned about relying on the dev environment to remain stable, and keeping dev in sync.

**Comment by jpetto** [comment]
going to merge this as the dev concerns, while valid, shouldn't be a blocker for this use case. and if they are, we can always revert/delete!

**Pull request merged**